Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp

```diff
@@ -465,14 +465,24 @@ void WebAutomationSession::switchToBrowsingContext(const Inspector::Protocol::Au
     ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!page, WindowNotFound);
 
     bool frameNotFound = false;
-    webFrameIDForHandle(frameHandle, frameNotFound);
+    auto frameID = webFrameIDForHandle(frameHandle, frameNotFound);
     ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(frameNotFound, FrameNotFound);
 
-    // FIXME: Should we also switch to the frame that frameHandle points to?
-    m_client->requestSwitchToPage(*this, *page, [page = Ref { *page }, callback = WTFMove(callback)]() {
+    m_client->requestSwitchToPage(*this, *page, [protectedThis = Ref { *this }, frameID, page = Ref { *page }, callback = WTFMove(callback)]() mutable {
         page->setFocus(true);
 
-        callback({ });
+        if (!frameID) {
+            callback({ });
+            return;
+        }
+
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!WebFrameProxy::webFrame(frameID.value()), FrameNotFound);
+
+        page->sendWithAsyncReplyToProcessContainingFrameWithoutDestinationIdentifier(frameID, Messages::WebAutomationSessionProxy::FocusFrame(page->webPageIDInMainFrameProcess(), frameID.value()), WTF::CompletionHandler<void(std::optional<String>&&)> { [callback = WTFMove(callback)] (std::optional<String>&& optionalError) mutable {
```

Comment:
NIT: can you just say `CompletionHandler { ... }` instead?  would be neat to be able to similar to how you can `Ref { ... }` without the type

Reply:
I just tried this but it complains that it can't deduce the template type arguments. :|

Reply:
Likewise the mutable seems to be needed because... I'm not quite sure really.

```
/app/webkit/WebKitBuild/GTK/Release/WTF/Headers/wtf/StdLibExtras.h: In instantiation of ‘constexpr typename std::remove_reference<_Arg>::type&& std::move(T&&) [with WTF::CheckMoveParameterTag <anonymous> = WTF::CheckMoveParameter; T = const WTF::Function<void(experimental::fundamentals_v3::expected<void, WTF::String>)>&; typename remove_reference<_Arg>::type = const WTF::Function<void(experimental::fundamentals_v3::expected<void, WTF::String>)>]’:
/app/webkit/Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:487:266:   required from here
/app/webkit/WebKitBuild/GTK/Release/WTF/Headers/wtf/StdLibExtras.h:839:51: error: static assertion failed: T is const qualified.
```

Reply:
https://ews-build.webkit.org/#/builders/2/builds/115744

Reply:
Okay, my current understanding is that because we WTFMove the `callback` out of the closure, the closures need to be marked `mutable` so that they are not inferred as `const`.